src/components/sortableList/index.tsx

```diff
@@ -1,44 +1,29 @@
 /* eslint-disable react-hooks/exhaustive-deps */
-import {map, mapKeys} from 'lodash';
+import {map, mapKeys, filter, reduce} from 'lodash';
 import React, {useMemo, useCallback} from 'react';
-import {FlatList, FlatListProps, LayoutChangeEvent} from 'react-native';
+import {FlatList, LayoutChangeEvent} from 'react-native';
 import {useSharedValue} from 'react-native-reanimated';
 import {GestureHandlerRootView} from 'react-native-gesture-handler';
-import SortableListContext, {SortableListContextType} from './SortableListContext';
+import SortableListContext from './SortableListContext';
 import SortableListItem from './SortableListItem';
 import {useDidUpdate, useThemeProps} from 'hooks';
+import {SortableListProps, SortableListItemProps} from './types';
+export {SortableListProps, SortableListItemProps};
 
-interface ItemWithId {
-  id: string;
-}
-
-export interface SortableListProps<ItemT extends ItemWithId>
-  extends Omit<FlatListProps<ItemT>, 'extraData' | 'data'>,
-    Pick<SortableListContextType, 'scale'> {
-  /**
-   * The data of the list, do not update the data.
-   */
-  data: FlatListProps<ItemT>['data'];
-  /**
-   * A callback to get the new order (or swapped items).
-   */
-  onOrderChange: (data: ItemT[] /* TODO: add more data? */) => void;
-  /**
-   * Whether to enable the haptic feedback
-   * (please note that react-native-haptic-feedback does not support the specific haptic type on Android starting on an unknown version, you can use 1.8.2 for it to work properly)
-   */
-  enableHaptic?: boolean;
+function generateItemsOrder<ItemT extends SortableListItemProps>(data: SortableListProps<ItemT>['data']) {
+  return map(data, item => item.id);
 }
 
-function generateItemsOrder<ItemT extends ItemWithId>(data: SortableListProps<ItemT>['data']) {
-  return map(data, item => item.id);
+function generateInertIds<ItemT extends SortableListItemProps>(data: SortableListProps<ItemT>['data']) {
+  return reduce(filter(data, 'inert'), (item, cur) => ({...item, [cur.id]: true}), {});
```

Comment:
I think I did that at first, I get a typescript error in `id` of `[cur.id]` which I can't solve ATM...

Reply:
Yes it's because of the `filter` method. 
For some reason its return value type is something weird instead of just an array items. 
maybe here `[cur.id]: true` you can do add casting `[(cur as ItemT).id]: true`, TBH, how can `cur` can be anything else?

Reply:
Worked 👍